aead/src/subtle/encrypt_then_authenticate.rs

```diff
@@ -81,7 +81,11 @@ impl tink_core::Aead for EncryptThenAuthenticate {
         let mut to_auth_data = Vec::with_capacity(additional_data.len() + ciphertext.len() + 8);
         to_auth_data.extend_from_slice(additional_data);
         to_auth_data.extend_from_slice(&ciphertext);
-        let aad_size_in_bits = (additional_data.len() * 8) as u64;
+        let aad_size_in_bits = (additional_data.len() as u64) * 8;
+        if (aad_size_in_bits / 8) != (additional_data.len() as u64) {
```

Comment:
We can use checked_mul of usize https://doc.rust-lang.org/std/primitive.usize.html#method.checked_mul
It's safe for overflow.